src/main/kotlin/com/exactpro/th2/kafka/client/MessageProcessor.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021-2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.kafka.client
+
+import com.exactpro.th2.common.grpc.RawMessage
+import com.exactpro.th2.common.grpc.RawMessageBatch
+import com.exactpro.th2.common.message.direction
+import com.exactpro.th2.common.message.sequence
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import io.reactivex.rxjava3.core.Flowable
+import io.reactivex.rxjava3.flowables.GroupedFlowable
+import io.reactivex.rxjava3.plugins.RxJavaPlugins
+import io.reactivex.rxjava3.processors.FlowableProcessor
+import io.reactivex.rxjava3.processors.UnicastProcessor
+import io.reactivex.rxjava3.subscribers.DisposableSubscriber
+import mu.KotlinLogging
+import java.io.Closeable
+import java.util.function.Supplier
+
+class MessageProcessor(
```

Comment:
I think you can use one from here instead
https://github.com/th2-net/th2-common-utils-j/blob/book-and-page/src/main/kotlin/com/exactpro/th2/common/utils/message/MessageBatcher.kt